### ISSUE TITLE
Provide access to the table properties to the WriterFactory

### ISSUE THREAD

**Issue by omalley** [Netflix/iceberg#15]
Currently, there isn't a way to get the table properties in the SparkOrcWriter via the WriterFactory.

**Comment by omalley** [comment]
The problem that I'm having is that I'd like the SparkOrcWriter to get the table properties without looking up the table again. In particular, it would be good to be able to set the compression, etc. via the table properties.

**Comment by rdblue** [comment]
Agreed, we should plan on passing the table properties through so all of the write configuration happens there.
